[DynamoDB] Update BulkSet and BulkDelete with BatchWriteItem

ksashikumar opened the pull request:
# Description

This PR updates `BulkSet` and `BulkDelete` for DynamoDB with [`BatchWriteItem`](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchWriteItem.html)

## Issue reference

This PR will close: https://github.com/dapr/components-contrib/issues/245

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dapr/components-contrib/331)
<!-- Reviewable:end -->

artursouza commented on state/aws/dynamodb/dynamodb.go:
```diff
@@ -102,13 +102,32 @@ func (d *StateStore) Set(req *state.SetRequest) error {
+						S: aws.String(r.Key),
+					},
+					"value": {
+						S: aws.String(fmt.Sprintf("%v", r.Value)),
```
I made this comment 9 days ago but forgot to publish. Can you validate this for other types? "S:" means always String.

ksashikumar replied:
@artursouza Thanks for the review 👍 I've updated it, could you please review again?

yaron2 commented:
@ksashikumar the tests are failing

artursouza commented:
@ksashikumar Unit tests are failing. Can you check it again?

ksashikumar commented:
@artursouza @yaron2 Sorry for the delay, I've fixed the tests and similar to `BulkSet` we should do the type validation for `state.GetRequest.Value` for `Set` as well. So I've cleaned up the code a bit and updated tests for `Set` also. Could you please review?

yaron2 commented:
Thanks for this @ksashikumar!
